examples/src/multiple_render_targets/mod.rs

```diff
@@ -0,0 +1,596 @@
+        (self.bindgroup_a, self.bindgroup_b) = Self::create_bindgroups(
+            device,
+            &self.bindgroup_layout,
+            texture_targets,
+            &self.sampler,
+        )
+    }
+}
+
+struct TextureTargets {
+    a_view: TextureView,
+    b_view: TextureView,
+}
+
+async fn run(event_loop: EventLoop<()>, window: Window) {
+    let mut size = window.inner_size();
+    size.width = size.width.max(1);
+    size.height = size.height.max(1);
+
+    let instance = wgpu::Instance::default();
+
+    let surface = instance.create_surface(&window).unwrap();
+    let adapter = instance
+        .request_adapter(&wgpu::RequestAdapterOptions {
+            power_preference: wgpu::PowerPreference::default(),
+            force_fallback_adapter: false,
+            // Request an adapter which can render to our surface
+            compatible_surface: Some(&surface),
+        })
+        .await
+        .expect("Failed to find an appropriate adapter");
+
+    // Create the logical device and command queue
+    let (device, queue) = adapter
+        .request_device(
+            &wgpu::DeviceDescriptor {
+                label: None,
+                required_features: wgpu::Features::empty(),
+                // Make sure we use the texture resolution limits from the adapter, so we can support images the size of the swapchain.
+                required_limits: wgpu::Limits::downlevel_webgl2_defaults()
+                    .using_resolution(adapter.limits()),
+            },
+            None,
+        )
+        .await
+        .expect("Failed to create device");
+
+    let mut config = surface
+        .get_default_config(&adapter, size.width, size.height)
+        .unwrap();
+    surface.configure(&device, &config);
+
+    let shader = device.create_shader_module(wgpu::ShaderModuleDescriptor {
+        label: None,
+        source: wgpu::ShaderSource::Wgsl(Cow::Borrowed(include_str!("shader.wgsl"))),
+    });
+
+    // Renderer that draws to 2 textures at the same time:
+    let mtr = MultiTargetRenderer::init(
+        &device,
+        &queue,
+        &shader,
+        // ColorTargetStates specify how the data will be written to the
+        // output textures:
+        &[
+            ColorTargetState {
+                format: config.format,
+                blend: None,
+                write_mask: Default::default(),
+            },
+            ColorTargetState {
+                format: config.format,
+                blend: None,
+                write_mask: Default::default(),
+            },
+        ],
+    );
+
+    // create our target textures that will receive the simultaneous rendering:
+    let mut texture_targets =
+        create_target_textures(&device, config.format, config.width, config.height);
+
+    // Helper renderer that displays the results in 2 separate viewports:
+    let mut drawer = TargetRenderer::init(&device, &shader, config.format, &texture_targets);
+
+    let window = &window;
+    event_loop
```

Comment:
Take a look at the other examples that are rendering to screen - we should use the `Framework` here, which not only streamlines testing, but means we don't need all this duplicate winit and execution code, and can focus on the wgpu code.